### DISCUSSION TITLE
Why can't we use modal with transition ?

### DISCUSSION THREAD

**Discussion by hieunguyentrung** [excid3/tailwindcss-stimulus-components#258] [Q&A]
https://github.com/excid3/tailwindcss-stimulus-components/blob/08e5776650d1b1c05b098d7109c64b37204e9705/src/modal.js#L23

Hello,
First, thank you for your hard work.
I'm implementing the modal and found out that it doesn't support transition.
I really want to use transition or something like that to improve UI/UX.

Please help me understand why the transition is not working and is there any way we can do it.
I am looking for an implementation to be able to use the below settings. Thank you.

```
Entering: "ease-out duration-300"
  From: "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
  To: "opacity-100 translate-y-0 sm:scale-100"
Leaving: "ease-in duration-200"
  From: "opacity-100 translate-y-0 sm:scale-100"
  To: "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95
```

**Comment by rgarver** [comment]
similar question for slideover.

**Comment by jaywhy** [comment]
I don't think that will work, because the slider and modal components use the [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) tag. In this library, transitions work by showing and hiding a div. Dialogs don't work that way.

The `dialog` tag has a `dialog:open` state, which can be styled and transitioned to. Check out the MDN article for some examples.
